src/components/Table/TableBasic.part.tsx

```diff
@@ -152,21 +152,31 @@ export class TableBasic<T> extends React.Component<TableProps<T> & RefProps, Tab
         row: -1,
       });
     } else if (this.isSortable(key, getColumns(data, columns))) {
-      const { sorting } = this.state;
-      const isAscending = sorting && sorting.order === 'descending' && sorting.columnKey === key;
-
-      if (!isAscending && column !== -1) {
-        this.setState({
-          sorting: {
-            columnKey: key,
-            order: sorting && sorting.columnKey === key ? 'descending' : 'ascending',
-          },
-        });
-      } else {
-        this.setState({
-          sorting: undefined,
-        });
-      }
+      this.setState(
+        ({ sorting }) => {
+          const isAscending = sorting && sorting.order === 'descending' && sorting.columnKey === key;
+          const order = sorting && sorting.columnKey === key ? 'descending' : 'ascending';
+          let newSortingValue: TableSorting | undefined = undefined;
+
+          if (!isAscending && column !== -1) {
+            newSortingValue = {
+              columnKey: key,
+              order,
+            };
+          }
+
+          return { sorting: newSortingValue };
+        },
+        () => {
+          if (typeof onSort === 'function') {
+            onSort({
+              column,
+              key,
+              order: this.state.sorting && this.state.sorting.order,
```

Comment:
You are destructuring the state on the param already, why do you need to call this.state again here?

Reply:
Because this is a differnt closure which has not access to the setState closure

Reply:
Oops, my bad. I didn't see it was the callback of the setState. :)